plugins/s3_auth/aws_auth_v4.h

```diff
@@ -67,7 +67,23 @@ static const String X_AMZ                = "x-amz-";
 static const String CONTENT_TYPE         = "content-type";
 static const String HOST                 = "host";
 
-String trimWhiteSpaces(const String &s);
+/**
+ * @brief Trim white spaces from beginning and end.
+ * @returns trimmed string
+ */
+template <class StringLike>
+StringLike
+trimWhiteSpaces(const StringLike &s)
+{
+  /* @todo do this better? */
+  static const auto whiteSpace = " \t\n\v\f\r";
+  size_t start                 = s.find_first_not_of(whiteSpace);
+  if (StringLike::npos == start) {
+    return StringLike();
+  }
+  size_t stop = s.find_last_not_of(whiteSpace);
+  return s.substr(start, stop - start + 1);
+}
```

Comment:
Since we're making this change, we might as well replace the uses of this with ts::TextView:
http://docs.solidwallofcode.com/libswoc/code/TextView.en.html

In that case, this function is removed and replaced with: 

```
ts::TextView tv{some_string};
tv.trim_if(&isspace);
```

That is, we should be able to entirely remove this function and replace all calls with ts::TextView.